### PR TITLE
HP-2255 replace CRA with Vite

### PR DESCRIPTION
**Pull request by mikkojamG**
Replacing react-scripts with Vite. Vite is currently the recommended bundler by React and also by our development guidelines. Also, getting rid of react-scripts will close [multiple vulnerabilities](https://github.com/City-of-Helsinki/open-city-profile-ui/security/dependabot).

- Uninstall react-scripts and replace with Vite. Add `vite.config.mts`.
- Updating development scripts, Vite handles environment variables a bit differently.
- Uninstall Jest and related dependencies, replace with Vitest. Fixing broken unit tests.
- Dockerfile changes
- README changes

### CODE REVIEW

**Comment by terovirtanen** [comment]
**HELSINKI-PROFILE-UI branch is deployed to platta: https://helsinki-profile-pr314-ui.dev.hel.ninja** :rocket::rocket::rocket:

**Comment by terovirtanen** [comment]
**HELSINKI-PROFILE-UI branch is deployed to platta: https://helsinki-profile-pr314-ui.dev.hel.ninja** :rocket::rocket::rocket:

**Comment by terovirtanen** [comment]
**HELSINKI-PROFILE-UI branch is deployed to platta: https://helsinki-profile-pr314-ui.dev.hel.ninja** :rocket::rocket::rocket:

**Comment by terovirtanen** [comment]
**HELSINKI-PROFILE-UI branch is deployed to platta: https://helsinki-profile-pr314-ui.dev.hel.ninja** :rocket::rocket::rocket:

**Comment by terovirtanen** [comment]
**HELSINKI-PROFILE-UI branch is deployed to platta: https://helsinki-profile-pr314-ui.dev.hel.ninja** :rocket::rocket::rocket:

**Comment by terovirtanen** [comment]
**HELSINKI-PROFILE-UI branch is deployed to platta: https://helsinki-profile-pr314-ui.dev.hel.ninja** :rocket::rocket::rocket:

**Comment by terovirtanen** [comment]
**HELSINKI-PROFILE-UI branch is deployed to platta: https://helsinki-profile-pr314-ui.dev.hel.ninja** :rocket::rocket::rocket:

**Comment by terovirtanen** [comment]
**HELSINKI-PROFILE-UI branch is deployed to platta: https://helsinki-profile-pr314-ui.dev.hel.ninja** :rocket::rocket::rocket:

File: scripts/update-runtime-env.ts [hunk 1]
```diff
@@ -1,30 +1,47 @@
...
 
+dotenv.config({
+  processEnv: import.meta.env,
+  ...(USE_TEST_ENV
+    ? { path: ['.env', '.env.test'] }
+    : { path: ['.env', `.env.${import.meta.env.NODE_ENV}`, '.env.local'] }),
+  override: true,
+});
+
+// Prevent collision is app is running while tests are started
```
Comment (Riippi) [inline scripts/update-runtime-env.ts]: is app is  -> if app is ?

File: scripts/update-translations.ts
Comment (arpargo) [inline scripts/update-translations.ts:1]: En ymmärrä näiden muutosten tarpeellisuutta suoralta kädeltä, voitko avata hieman?

**Comment by terovirtanen** [comment]
**HELSINKI-PROFILE-UI branch is deployed to platta: https://helsinki-profile-pr314-ui.dev.hel.ninja** :rocket::rocket::rocket:

**Review by Riippi — CHANGES_REQUESTED** [review]

File: scripts/update-runtime-env.ts (hunk 1 above)
Comment (mikkojamG) [reply scripts/update-runtime-env.ts]: Fixed: https://github.com/City-of-Helsinki/open-city-profile-ui/pull/314/commits/887d9694a491b33d0e4c756d7689a0406c449268

**Review by Riippi — APPROVED** [review]
Great success 👍

**Comment by terovirtanen** [comment]
**HELSINKI-PROFILE-UI branch is deployed to platta: https://helsinki-profile-pr314-ui.dev.hel.ninja** :rocket::rocket::rocket:

**Review by arpargo — APPROVED** [review]
LGTM

**Comment by terovirtanen** [comment]
**HELSINKI-PROFILE-UI branch is deployed to platta: https://helsinki-profile-pr314-ui.dev.hel.ninja** :rocket::rocket::rocket:

**Pull request merged**